File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsRolesPermissionsAdmin.swift
import Foundation

// MARK: - Roles & permissions admin (M14.2)

struct RBACPermission: Decodable, Identifiable, Hashable {
    var id: String
    var permissionString: String
    var description: String
    var createdAt: String?
}

struct RoleWithPermissions: Decodable, Identifiable, Hashable {
    var id: String
    var name: String
    var description: String?
    var scope: String?
    var createdAt: String?
    var permissions: [RBACPermission]
}

struct RBACUserBrief: Decodable, Identifiable, Hashable {
    var id: String
    var email: String
    var displayName: String?
    var sid: String?
}

struct RolesListResponse: Decodable {
    var roles: [RoleWithPermissions]
}

struct RoleUsersResponse: Decodable {
    var users: [RBACUserBrief]
}

struct AddRoleUserRequest: Encodable {
    var userId: String
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsRolesPermissionsAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class RbacPermission(
    val id: String,
    val permissionString: String,
    val description: String = "",
    val createdAt: String? = null,
)

@Serializable
data class RoleWithPermissions(
    val id: String,
    val name: String,
    val description: String? = null,
    val scope: String? = null,
    val createdAt: String? = null,
    val permissions: List<RbacPermission> = emptyList(),
)

@Serializable
data class RbacUserBrief(
    val id: String,
    val email: String,
    val displayName: String? = null,
    val sid: String? = null,
)

@Serializable
data class RolesListResponse(
    val roles: List<RoleWithPermissions> = emptyList(),
)

@Serializable
data class RoleUsersResponse(
    val users: List<RbacUserBrief> = emptyList(),
)

@Serializable
data class AddRoleUserRequest(
    val userId: String,
)
